components/camera/replaypcd/replaypcd.go

```diff
@@ -26,41 +38,110 @@ func init() {
 }
 
 // TimeInterval holds the start and end time used to filter data.
 type TimeInterval struct {
-	Start time.Time `json:"start,omitempty"`
-	End   time.Time `json:"end,omitempty"`
+	Start string `json:"start,omitempty"`
+	End   string `json:"end,omitempty"`
 }
 
 // Validate checks that the config attributes are valid for a replay camera.
-func (c *Config) Validate(path string) ([]string, error) {
-	return nil, nil
+func (cfg *Config) Validate(path string) ([]string, error) {
+	if cfg.Source == "" {
+		return nil, goutils.NewConfigValidationFieldRequiredError(path, "source")
+	}
+
+	if cfg.Interval.Start != "" && cfg.Interval.Start > time.Now().UTC().Format(timeFormat) {
+		return nil, errors.New("invalid config, start time must be in the past")
+	}
+
+	if cfg.Interval.End != "" && cfg.Interval.End > time.Now().UTC().Format(timeFormat) {
+		return nil, errors.New("invalid config, end time must be in the past")
+	}
+
+	if cfg.Interval.Start != "" && cfg.Interval.End != "" && cfg.Interval.Start > cfg.Interval.End {
+		return nil, errors.New("invalid config, end time must be after start time")
+	}
+
+	return []string{cloud.InternalServiceName.String()}, nil
 }
 
 // pcdCamera is a camera model that plays back pre-captured point cloud data.
 type pcdCamera struct {
 	resource.Named
 	logger golog.Logger
+
+	cloudConnSvc cloud.ConnectionService
+	cloudConn    rpc.ClientConn
+	dataClient   datapb.DataServiceClient
+
+	source  string
+	robotID string
+
+	lastData string
+	limit    uint64
+	filter   *datapb.Filter
 }
 
 // newPCDCamera creates a new replay camera based on the inputted config and dependencies.
-func newPCDCamera(ctx context.Context, deps resource.Dependencies, cfg resource.Config, logger golog.Logger) (camera.Camera, error) {
+func newPCDCamera(ctx context.Context, deps resource.Dependencies, conf resource.Config, logger golog.Logger) (camera.Camera, error) {
 	cam := &pcdCamera{
+		Named:  conf.ResourceName().AsNamed(),
 		logger: logger,
+		limit:  1,
 	}
-	// TODO: Add start protocol for replay camera https://viam.atlassian.net/browse/RSDK-2893
+
+	if err := cam.Reconfigure(ctx, deps, conf); err != nil {
+		return nil, err
+	}
+
 	return cam, nil
 }
 
-// NextPointCloud is part of the camera interface but is not implemented for replay.
+// NextPointCloud returns a point cloud retrieved from app based on the applied filter.
 func (replay *pcdCamera) NextPointCloud(ctx context.Context) (pointcloud.PointCloud, error) {
-	var pc pointcloud.PointCloud
-	return pc, errors.New("NextPointCloud is unimplemented")
+	resp, err := replay.dataClient.BinaryDataByFilter(ctx, &datapb.BinaryDataByFilterRequest{
+		DataRequest: &datapb.DataRequest{
+			Filter:    replay.filter,
+			Limit:     replay.limit,
+			Last:      replay.lastData,
+			SortOrder: datapb.Order_ORDER_ASCENDING,
+		},
+		CountOnly:     false,
+		IncludeBinary: true,
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	// If no data is returned, return an error.
```

Comment:
We specified in the tech spec: "If no end time is specified, we will continue to poll for new data and the point cloud stream will not have an endpoint."

Reply:
Got it, that makes sense.

Reply:
Can you please update this in the tech spec?

Reply:
Updated!

Reply:
thanks!